### ISSUE TITLE
Feature: overview of mail policies and their order

### ISSUE THREAD

**Issue by jbostoen** [jbostoen/itop-jb-mail-to-ticket-automation-v2#29]
It might be nice to have a short visual overview of the mail policies and their order, without having to figure it out manually by looking at the XML or checking the debug trace.

**Comment by jbostoen** [comment]
Won't do this for now, since some steps are applied internally and they have no proper dictionary entry. For a typical administrator who uses this extension, there's no added value.

**Comment by jbostoen** [comment]
Revisiting this — the codebase has changed a lot since the 2022 "won't do" comment, and I think the original objection ("some steps are applied internally and have no proper dictionary entry") can now be solved cheaply.

**Why it's more feasible today:** "Policies" have been replaced by the `Steps/*` architecture (see `documentation/upgrade notes.md`). Every step is a self-describing class:
- `$iPrecedence` (execution order) and `GetXMLSettingsPrefix()` (its config prefix on the mailbox) are already public/introspectable (`Steps/Base.php`).
- `ProcessingHelper::GetAvailableSteps()` / `GetApplicableSteps()` already do exactly the discovery + filtering (by mailbox behavior) needed to build an overview — no new registry required.
- Every step class already carries a one- or two-line docblock summary (e.g. `AutoReply`: "A policy to handle auto-reply e-mails.") — this is developer documentation that already exists for free, and sidesteps the "no dictionary entry" objection entirely: no new translations would be required to ship this.

**Proposed user-friendly implementation:**
- A read-only tab/page on the mailbox object (similar in spirit to the existing mailbox monitoring page in `jb-email-synchro/ajax.php`), e.g. "Processing order" — listing, for *that specific mailbox*, only the steps that are actually applicable to it (via `GetApplicableSteps()`, so administrators don't see internal steps that don't apply to their config).
- Per row: step short name, its precedence number, an active/inactive badge derived from the mailbox's own `<prefix>_behavior` setting, and a one-line description pulled via `ReflectionClass::getDocComment()` off the step class (falling back to the class's short name if a step has no docblock, so nothing ever renders blank).
- Purely a read view (no editing) — administrators still configure behavior via the existing mailbox fields; this just answers "what runs, in what order, for this mailbox" without reading XML or turning on the debug trace, which was the original ask.
- Because it's built entirely from `GetApplicableSteps()` + reflection, it stays correct automatically as steps are added/removed/reordered — no maintenance burden per new step.

This keeps the "no added value / no dictionary entries" concern from 2022 fully addressed while giving admins the visual overview they'd find useful. Let me know if this sounds like the right shape and I'll flesh out an implementation.

**Comment by jbostoen** [comment]
Follow-up on feasibility: `IsApplicable()` is not purely mailbox-dependent — it's a bare static method with no parameters, and it can read whatever's currently loaded into `ProcessingHelper`'s statics: `GetMailBox()`, but also `GetMail()` and `GetTicket()`. The doc comment on `IsApplicable()` even hints at this ("Make sure the initiation of the step has already happened!").

Today only one step overrides it (`MatchByInReplyToOrReferences`, gated on a global setting, not mail content), which is why calling `GetApplicableSteps()` outside of real processing (e.g. from a static mailbox tab) happens to work — nothing currently dereferences `GetMail()`/`GetTicket()`. But `GetMail()` returns a non-nullable typed `EmailMessage` property. If a future step's `IsApplicable()` ever checks something mail-specific (e.g. a `NonDeliveryReport`-style step deciding applicability from the message itself), calling it from a page where no email is loaded would throw a `TypeError` rather than just being wrong.

So the original plan (build the overview purely from `GetApplicableSteps()` on the mailbox) is correct today but fragile — it silently assumes no step's `IsApplicable()` ever looks at the mail, which nothing in the framework enforces.

**Revised visualization approach**, split into two sections instead of one table that overpromises:
- **"Configured for this mailbox"** — structural only: `GetAvailableSteps()` filtered by `<prefix>_behavior != inactive`, skipping the `IsApplicable()` call entirely. Always safe to render on the mailbox tab, no email context needed.
- **"Applicable to a specific message"** (optional, opt-in) — let the admin pick a sample `EmailMessage`/`Ticket` (or an already-processed one), `SetMail()`/`SetTicket()` it into `ProcessingHelper`, then run the real `GetApplicableSteps()` and show the actual per-step verdict for that one message.

Optionally, use reflection to badge any step whose `IsApplicable()` is overridden (i.e. not inherited from `Base`) as "context-dependent" in the structural list, so admins know its true/false there isn't guaranteed without a real message.
